iceoryx_posh/include/iceoryx_posh/internal/popo/ports/publisher_port_data.hpp

```diff
@@ -50,6 +50,9 @@ struct PublisherPortData : public BasePortData
         ChunkSenderData<MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY, ChunkDistributorData_t>;
 
     ChunkSenderData_t m_chunkSenderData;
+
+    PublisherOptions m_options;
```

Comment:
While I think it is a good idea to save this in the port in order to be able to do the matching, I'm a little bit hesitant that we run into issues since some of the options are additionally stored at a other locations. It might happen that they are mutated and then out of sync. Maybe a issue to check if this duplication can be removed would be good.

Reply:
@MatthiasKillat yes, I mean a new issue, just to not forget about this potential issue. If not before v2 we should definitely look afterwards on this and consolidate the options to a single source of truth.

Reply:
cf. https://github.com/eclipse-iceoryx/iceoryx/issues/1048